### PR TITLE
Add `android_debug_unopt_arm64` to `local_engine.json`.

### PR DESCRIPTION
**Pull request by matanlurey**
As written:

```sh
flutter % et query builders | grep unopt_arm64                           
   "android_debug_unopt_arm64" config
   "host_debug_unopt_arm64" config
```

Yay!

### CODE REVIEW

File: ci/builders/local_engine.json [hunk 1]
```diff
@@ -19,6 +19,26 @@
         "targets": []
       }
     },
+    {
+      "drone_dimensions": [
```
Comment (zanderso) [inline ci/builders/local_engine.json:23]: I hope to get the builds in this file running on CI soon, so wondering if local workflows will need both the unopt and non-unopt version of android_debug_arm64?

Comment (matanlurey) [reply ci/builders/local_engine.json:23]: We talked offline, local builds will _probably_ all be `unopt` (the other configurations will be covered by release builders), so I am going to remove `android_debug_arm64` (/cc @johnmccutchan) from this file, and then continue to chat with John about "what should we do for duplicate names in general".

Comment (johnmccutchan) [reply ci/builders/local_engine.json]: I only use android_debug_arm64 and never the _unopt_ variant. So please don't remove it.

**Review by johnmccutchan — CHANGES_REQUESTED** [review]
This will break my workflow

File: ci/builders/local_engine.json (hunk 1 above)
Comment (matanlurey) [reply ci/builders/local_engine.json]: > I only use android_debug_arm64 and never the _unopt_ variant. So please don't remove it.

Ack. Can I clarify though, why is [this builder](https://github.com/flutter/engine/blob/622b37282f20e094cf43ff9f7485305199840121/ci/builders/linux_android_debug_engine.json#L118) not sufficient?

Comment (matanlurey) [reply ci/builders/local_engine.json:23]: Chatted offline, because that builder doesn't run/allow `macos`. Reverted that `-` block.

**Review by zanderso — APPROVED** [review]

**Pull request merged**